src/BenchmarkDotNet/Exporters/PrettyGithubMarkdownDiffDisassemblyExporter.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using BenchmarkDotNet.Diagnosers;
+using BenchmarkDotNet.Loggers;
+using BenchmarkDotNet.Reports;
+using BenchmarkDotNet.Running;
+using StreamWriter = BenchmarkDotNet.Portability.StreamWriter;
+
+namespace BenchmarkDotNet.Exporters
+{
+    public class PrettyGithubMarkdownDiffDisassemblyExporter : PrettyGithubMarkdownDisassemblyExporterBase
+    {
+        private readonly IReadOnlyDictionary<BenchmarkCase, DisassemblyResult> results;
+
+        private static bool canRead;
+
+        protected override string FileExtension => "md";
+        protected override string FileCaption => "asm.pretty.diff";
+
+        public PrettyGithubMarkdownDiffDisassemblyExporter(IReadOnlyDictionary<BenchmarkCase, DisassemblyResult> results)
+        {
+            this.results = results;
+        }
+
+        public override void ExportToLog(Summary summary, ILogger logger)
+        {
+            var benchmarksCases = summary.BenchmarksCases
+                .Where(results.ContainsKey).ToList();
+
+            logger.WriteLine($"## {summary.Title}");
+
+            for (int i = 0; i < benchmarksCases.Count; i++)
+            {
+                var firstBenchmarkCase = benchmarksCases[i];
+                for (int j = i + 1; j < benchmarksCases.Count; j++)
+                {
+                    var secondBenchmarkCase = benchmarksCases[j];
+
+                    var firstFileName = Export(summary, results[firstBenchmarkCase]);
+                    var secondFileName = Export(summary, results[secondBenchmarkCase]);
+                    try
+                    {
+                        var builder = new StringBuilder();
+
+                        RunGitDiff(firstFileName, secondFileName, builder);
+
+                        if (firstBenchmarkCase.Descriptor.WorkloadMethod == secondBenchmarkCase.Descriptor.WorkloadMethod) // diff between the same method for different JITs
```

Comment:
I think that this is the only case when we should be producing the diff if we want to have it enabled by default.

Example: a `ListBenchmarks` type with 3 benchmarks: `Add`, `AddRange` and `Insert`. When we use the disassembly diagnoser we want to get the disassembly for all 3 of them. But we are rather not interested in diff. 
When we run it for few different Runtime settings, then we want to get the diff.